services/src/panic.c

```diff
@@ -16,89 +26,79 @@
 #include "interrupts_hal.h"
 #include "hal_platform.h"
 #include "core_hal.h"
+#include "delay_hal.h"
 
-LOG_SOURCE_CATEGORY("panic");
-
-#define LOOPSPERMSEC 5483
+#define PANIC_LED_COLOR RGB_COLOR_RED
 
-typedef struct {
-        uint32_t  led;
-        uint16_t  count;
-} flash_codes_t;
+LOG_SOURCE_CATEGORY("panic");
 
+static void panic_internal(const ePanicCode code, const void* extraInfo);
 
-#define def_panic_codes(_class,led,code) {led, code},
-static const flash_codes_t flash_codes[] = {
-                 {0,0}, //NotUsedPanicCode
-#include "panic_codes.h"
-};
-#undef def_panic_codes
+static PanicHook panicHook = panic_internal;
```

Comment:
This should be `nullptr` after refactoring, otherwise we'll run `panic_internal` twice.